plugins/file/plugin.go

```diff
@@ -113,10 +165,25 @@ func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
 
 // Handler4 handles DHCPv4 packets for the file plugin
 func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
-	// TODO check the MAC address in the request
-	//      if it is present in StaticRecords, forge a response
-	//      and stop processing.
-	return resp, true
+	record, ok := DHCPv4Records[req.ClientHWAddr.String()]
+	if !ok {
+		log.Printf("plugins/file: MAC address %s is new, leasing new IP address", req.ClientHWAddr.String())
+		rec, err := createIP(ipRangeStart, ipRangeEnd)
```

Comment:
this means that the IP address is picked in a range, while the `file` plugin is meant for static file mappings. This should be implemented in a different plugin, e.g. something called `range` or similar name.

Reply:
If I were to implement another plugin called range that creates a new IP and leases it, that plugin would still have to open the leases file, check if the IP is taken and save it after it was leased. Then the file plugin and the range plugin would be very similar. Is that something you want? If so, then the range plugin would accept the arguments the file plugin accepts now, and the file plugin would accept the filename and a lease-time so it can expand the lease if it expired once it leases an IP address? Furthermore, file plugin would be second to last in the order of plugins and return false if it found an existing IP, and the range plugin would be the last plugin and would handle the request only if the file plugin returned false?

Reply:
I don't have a strong opinion about it. The plugins will share some logic. Maybe we can have them separate for now, and at some point if it makes sense we can merge them

Reply:
As long as the plug-in behaviour is consistent (i.e. v4 and v6 do the same things where it makes sense) and the name and documentation match what the plug-in does, I'm fine with it.
In this case the file plugin is for static file mapping (like a database), and meant to be read only. I'm ok with changing it (and renaming) to something more useful in the real world, like the range plugin discussed above, and if we do so, the v6 code has to behave the same way.

To sum up, what about renaming it to `ramge` and do the same with v6?

Regarding static mapping, I had in mind writing a sqlite plugin for quite some time, that could do what `file` does today before your change